### ISSUE TITLE
Speeding up running the tests

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#1708]
Here is a test:
```m2
i1 : elapsedTime check_1 "Macaulay2Doc"
-* running test 1 of package Macaulay2Doc in file:
   ../../Macaulay2/packages/Macaulay2Doc/doc9.m2:230:1:
   rerun with: check_1 "Macaulay2Doc" *-
 -- 9.86182 seconds elapsed
```
And the content of the test that took almost 10 seconds? Here it is:
https://github.com/Macaulay2/M2/blob/afaa83b5c6062dcbfc7f383bf4786d17ed0a0c3d/M2/Macaulay2/packages/Macaulay2Doc/doc9.m2#L227-L229
This is _objectively_ bad.

Here is the reason: when `(check, ZZ, String)` is called, two things happen:
1. `needsPackage (pkg, LoadDocumentation => true)`
https://github.com/Macaulay2/M2/blob/afaa83b5c6062dcbfc7f383bf4786d17ed0a0c3d/M2/Macaulay2/m2/testing.m2#L46
2. `loadPackage(pkg#"pkgname", LoadDocumentation => true, Reload => true)`
https://github.com/Macaulay2/M2/blob/afaa83b5c6062dcbfc7f383bf4786d17ed0a0c3d/M2/Macaulay2/m2/testing.m2#L31

Loading the documentation is necessary only because some authors have added package tests after `beginDocumentation()`. I see the appeal of this, especially since you can clearly put the tests near their respective documentation:
```m2
document{ key => "2 + 2",
    EXAMPLE "2 + 2",
    TEST "assert(2 + 2 = 4)"}
```
However, at least for `Macaulay2Doc`, this is slowing down running the tests **very** significantly, as the example above demonstrates.

Here are the two solutions I see:
1. use `capture` for tests instead (see #1689)
2. remove `LoadDocumentation => true` in `check` and enforce that all `TEST`s must appear before `beginDocumentation`
3. do one of the suggestions above, but only for Macaulay2Doc

@mikestillman @DanGrayson any thoughts on either of these?

This is somewhat a companion to #1480, as it speeds up testing of Macaulay2Doc very very significantly!

**Comment by mahrud** [comment]
Also relevant to this discussion: 

> Perhaps all the `Core` tests should be merged into the same place?  Macaulay2Doc doesn't test everything.  For example, for the new `Core` functions I've contributed recently (`regexQuote`, `findProgram` and `runProgram`), I only wrote tests for `M2/Macaulay2/tests/normal`.  Macaulay2Doc got examples, of course, but no additional tests.

_Originally posted by @d-torrance in https://github.com/Macaulay2/M2/issues/1547#issuecomment-712265915_

**Comment by DanGrayson** [comment]
> Here are the two solutions I see:
> 
> 1. use `capture` for tests instead (see #1689)
> 2. remove `LoadDocumentation => true` in `check` and enforce that all `TEST`s must appear before `beginDocumentation`
> 3. do one of the suggestions above, but only for Macaulay2Doc
> 
> @mikestillman @DanGrayson any thoughts on either of these?
> 

Let's not disturb users with this issue.  Let's disallow the use of TEST completely in Macaulay2Doc.  Just move all those tests to M2/Macaulay2/tests.

**Comment by mahrud** [comment]
Hmm, this is done, but I did not get the speedup I expected 🤔 I think I missed something in my explanation above: only the first test needs to load all of the documentation, after that it's the same speed.

There is still some speedup because when possible starting a new process is avoided, but for some reason this works locally and not on github: https://github.com/Macaulay2/M2/pull/1709#issuecomment-748269657

**Comment by mahrud** [comment]
After #1716, the following are the only packages where capture fails on at least one test: (13 test total)
```
check("EllipticIntegrals")
check("InvariantRing")
check("MixedMultiplicity")
check(20, "Schubert2")
```
Also, the following packages are the only ones where the runtime is slower by more than 5 seconds:
```m2
(4.98, Saturation),
(7.35, SpecialFanoFourfolds),
(7.98, ToricVectorBundles),
(8.51, OldToricVectorBundles),
(14.87, QthPower),
(111.99, Polyhedra)
```
